core/templates/dev/head/pages/load_constants.js

```diff
@@ -0,0 +1,19 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var request = new XMLHttpRequest();
+request.overrideMimeType('application/json');
+request.open('GET', '/assets/constants.json', false);
+request.send(null);
+var CONSTANTS = JSON.parse(request.responseText);
```

Comment:
I have a few things to mention here. We have at least two choices for doing this part. Either we can have one top-level CONSTANTS object which contains all the constants, or we can have individual constants for each element in the CONSTANTS object. I'm not actually sure which I prefer. @seanlip what do you think?

Regarding having a single CONSTANTS objects, we could instead use a provider (https://thinkster.io/egghead/providers) and avoid the ``oppia.constant()`` used earlier. The provider's ``$get`` would be used to retrieve the constants and always return a new copy of them (to avoid mutation, effectively making CONSTANTS constant).

If we wanted each constant to be its own value, then in this file we could retrieve the constants, iterate across its keys, then call ``oppia.constant()`` for each one. We need to make sure we only call this once, which is slightly tricky because we are moving away from having a base html file.

Regarding actually retrieving the constants, you might be able to do something like in http://stackoverflow.com/questions/13515141/html-javascript-how-to-access-json-data-loaded-in-a-script-tag-with-src-set where we load the JSON file as a script tag, then retrieve the values using DOM access (no needed for the unnecessary jquery). It avoids having to write out a XMLHttpRequest, at least.